algorithms/HPdutility/man/hpdsample.Rd

```diff
@@ -0,0 +1,85 @@
+\name{hpdsample}
+\alias{hpdsample}
+\title{A sampling algorithm for darrays/dframes}
+\description{
+    Samples data from data1 (and when present, data2, maintaining the same
+    order of samples between them) into a new darray/dframe with nSamplePartitions
+    partitions, each of which contains approximately samplingRatio * nrow(data1)
+    elements.
+}
+\usage{
+hpdsample(data1, data2, nSamplePartitions, samplingRatio, trace = FALSE)
+}
+
+\arguments{
+  \item{data1}{
+      A darray/dframe to sample
+  }
+  \item{data2}{
+      (Optional) A darray/dframe to sample along with data1. Must have the same number 
+      of rows and partitions as data1. Each row of data2 is assumed to correspond 
+      with the same row of data1, and this correspondence will be maintained
+      when sampling.
+  }
+  \item{nSamplePartitions}{
+      The number of output partitions for the sample data
+  }
+  \item{samplingRatio}{
+      The fraction of data from data1/data2 to be sampled
+      in each output partition. E.g. if nrow(data1) = 100,
+      and samplingRatio is 0.1, each output partition will
+      contain approximately 100 * 0.1 = 10 rows.
+  }
+  \item{trace}{
+      A boolean that states whether or not to print the
+      progress of the distributed foreach calls
```

Comment:
Please remove the end of the sentence, "of the distributed foreach calls". We may add more tracing messages in the future.